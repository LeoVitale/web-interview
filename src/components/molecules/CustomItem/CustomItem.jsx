import React from 'react'
import PropTypes from 'prop-types'
import Avatar from 'components/atoms/Avatar'
import UserHeader from 'components/atoms/UserHeader'
import Icon from 'components/atoms/Icon'
import {
  userItem,
  userData,
  actions,
  addIcon,
  addAvatar,
} from './CustomItem.module.scss'

const CustomItem = ({ action, header, avatar, subHeader }) => {
  const userProps = { title: header, subHeader }

  const avatarProps = avatar
    ? { name: header, img: avatar }
    : { className: addAvatar, icon: <Icon type="add" className={addIcon} /> }

  return (
    <div className={userItem}>
      <div className={userData}>
        <Avatar size="small" {...avatarProps} />
        <UserHeader {...userProps} />
      </div>
      <div className={actions}>{action}</div>
    </div>
  )
}

CustomItem.propTypes = {
  action: PropTypes.node,
  user: PropTypes.object,
  subHeader: PropTypes.string,
}

CustomItem.defaultProps = {
  user: {},
  subHeader: '',
}

export default CustomItem
