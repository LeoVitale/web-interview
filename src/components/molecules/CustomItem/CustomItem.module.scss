@import 'styles/_constants.scss';

.userItem {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 20px;
  .userData {
    display: flex;
    align-items: center;
  }
  .addAvatar {
    background-color: $primary;
  }
  .addIcon {
    fill: white;
  }
}
