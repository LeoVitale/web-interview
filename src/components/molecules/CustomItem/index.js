export { default } from './CustomItem.jsx'
