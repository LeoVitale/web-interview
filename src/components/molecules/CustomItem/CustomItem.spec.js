import React from 'react'
import { mount } from 'enzyme'
import CustomItem from './CustomItem.jsx'

describe('<CustomItem />', () => {
  const customItem = mount(<CustomItem />)

  it('should render CustomItem as expected', () => {
    expect(customItem).toMatchSnapshot()
  })
})
