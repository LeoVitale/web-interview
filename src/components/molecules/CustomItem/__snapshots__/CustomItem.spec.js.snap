// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomItem /> should render CustomItem as expected 1`] = `
<CustomItem
  subHeader=""
  user={Object {}}
>
  <div
    className="userItem"
  >
    <div
      className="userData"
    >
      <Avatar
        className="addAvatar"
        icon={
          <Icon
            className="addIcon"
            type="add"
          />
        }
        img=""
        name=""
        size="small"
      >
        <div
          className="avatar addAvatar small"
        >
          <Icon
            className="addIcon"
            type="add"
          >
            <svg
              className="svgIcon addIcon"
              fill="#400099"
              height="24"
              role="img"
              viewBox="0 0 24 24"
              width="24"
            >
              <title />
              <path
                d="M12.75 12.75v6.513a.75.75 0 1 1-1.5 0V12.75H4.74a.75.75 0 1 1 0-1.5h6.51V4.713a.75.75 0 1 1 1.5 0v6.537h6.568a.75.75 0 1 1 0 1.5H12.75z"
              />
            </svg>
          </Icon>
        </div>
      </Avatar>
      <UserHeader
        subHeader=""
        subheader=""
        title=""
      >
        <div />
      </UserHeader>
    </div>
    <div
      className="actions"
    />
  </div>
</CustomItem>
`;
